pkg/api/handlers/mcp.go

```diff
@@ -1231,34 +1189,79 @@ func addExtractedEnvVarsToCatalogEntry(entry *v1.MCPServerCatalogEntry) {
-		for _, env := range entry.Spec.URLManifest.Env {
-			existingURL[env.Key] = struct{}{}
-		}
+// addExtractedEnvVarsToCatalogEntry extracts and adds environment variables to the catalog entry manifest
+func addExtractedEnvVarsToCatalogEntry(entry *v1.MCPServerCatalogEntry) {
+	// Keep track of existing env vars in the manifest to avoid duplicates
+	existing := make(map[string]struct{})
+	for _, env := range entry.Spec.Manifest.Env {
+		existing[env.Key] = struct{}{}
+	}
 
-		// Extract variables from URL
-		extractedURL := make(map[string]struct{})
-		for _, v := range extractEnvVars(entry.Spec.URLManifest.FixedURL) {
-			extractedURL[v] = struct{}{}
-		}
+	// Extract variables based on runtime type
+	extracted := make(map[string]struct{})
 
-		// Add any new vars to the URL Manifest's Env list
-		for v := range extractedURL {
-			if _, exists := existingURL[v]; !exists {
-				entry.Spec.URLManifest.Env = append(entry.Spec.URLManifest.Env, types.MCPEnv{
-					MCPHeader: types.MCPHeader{
-						Name:        v,
-						Key:         v,
-						Description: "Automatically detected variable",
-						Sensitive:   true,
-						Required:    true,
-					},
-				})
+	switch entry.Spec.Manifest.Runtime {
+	case types.RuntimeUVX:
+		if entry.Spec.Manifest.UVXConfig != nil {
+			// Extract variables from command
+			for _, v := range extractEnvVars(entry.Spec.Manifest.UVXConfig.Command) {
+				extracted[v] = struct{}{}
+			}
+
+			// Extract variables from args
+			for _, arg := range entry.Spec.Manifest.UVXConfig.Args {
+				for _, v := range extractEnvVars(arg) {
+					extracted[v] = struct{}{}
+				}
+			}
+		}
+	case types.RuntimeNPX:
+		if entry.Spec.Manifest.NPXConfig != nil {
+			// Extract variables from args
+			for _, arg := range entry.Spec.Manifest.NPXConfig.Args {
+				for _, v := range extractEnvVars(arg) {
+					extracted[v] = struct{}{}
+				}
+			}
+		}
+	case types.RuntimeContainerized:
+		if entry.Spec.Manifest.ContainerizedConfig != nil {
+			// Extract variables from command
+			for _, v := range extractEnvVars(entry.Spec.Manifest.ContainerizedConfig.Command) {
+				extracted[v] = struct{}{}
 			}
+			// Extract variables from args
+			for _, arg := range entry.Spec.Manifest.ContainerizedConfig.Args {
+				for _, v := range extractEnvVars(arg) {
+					extracted[v] = struct{}{}
+				}
+			}
+		}
+	case types.RuntimeRemote:
+		if entry.Spec.Manifest.RemoteConfig != nil {
+			// Extract variables from FixedURL if present
+			if entry.Spec.Manifest.RemoteConfig.FixedURL != "" {
+				for _, v := range extractEnvVars(entry.Spec.Manifest.RemoteConfig.FixedURL) {
+					extracted[v] = struct{}{}
+				}
+			}
+		}
+	}
+
+	// Add any new vars to the manifest's Env list
+	for v := range extracted {
+		if _, exists := existing[v]; !exists {
+			entry.Spec.Manifest.Env = append(entry.Spec.Manifest.Env, types.MCPEnv{
+				MCPHeader: types.MCPHeader{
+					Name:        v,
+					Key:         v,
+					Description: "Automatically detected variable",
```

Comment:
what does Automatically detected variable mean?

Reply:
This isn't new, just moved from somewhere else.

When we detect things inside of `${}` in some of the configuration fields, we interpolate those as environment variables, and have to add them to the server manifest as something for the user to fill in.